lib/tasks/cap_cutover.rake

```diff
@@ -102,363 +33,13 @@ namespace :cap_cutover do
-
-      GC.start if total_running_count % 5000 == 0
-
-      if total_running_count % 1000 == 0
-        Rails.logger.debug "#{total_running_count} in #{distance_of_time_in_words_to_now(start_time, true)}"
-        @cap_import_logger.info total_running_count.to_s + ' in ' + distance_of_time_in_words_to_now(start_time, true)
-      end
-    end
-    @cap_import_logger.info 'Finished import.' + Time.zone.now.to_s
-    @cap_import_logger.info lines.to_s + ' lines of file: ' + args.file_location.to_s + ' were processed.'
-    @cap_import_logger.info total_running_count.to_s + 'records were processed in ' + distance_of_time_in_words_to_now(start_time, true)
-  end
-
-  desc 'create publication and contribution records from full cap delta'
-  task :create_pubs_from_delta, [:file_location] => :environment do |_t, args|
-    include ActionView::Helpers::DateHelper
-    start_time = Time.zone.now
-    total_running_count = 0
-    @cap_import_logger = Logger.new(Rails.root.join('log', 'cap_create_pmid_pubs.log'))
-    @cap_import_logger.info 'Started cap build pub process ' + Time.zone.now.to_s
-
-    lines = CSV.foreach(args.file_location, headers: true, header_converters: :symbol) do |row|
-      begin
-        total_running_count += 1
-
-        university_id = row[:university_id]
-        sunetid = row[:sunetid]
-        author = nil
-        unless university_id.nil? || university_id.blank? || university_id.to_s == '0'
-          author = Author.where(university_id: university_id).first
-        end
-        if author.nil? && !sunetid.nil? && !sunetid.blank?
-          author = Author.where(sunetid: sunetid).first
-        end
-        if author.nil?
-          @cap_import_logger.warn("Author not found- univid: #{university_id} sunetid: #{sunetid}")
-          next
-        end
-
-        pub = Publication.where(pmid: row[:pubmed_id]).first_or_create
-        pub.contributions.where(author_id: author.id).first_or_create(
-          cap_profile_id: author.cap_profile_id,
-          status: row[:authorship_status],
-          visibility: row[:visibility],
-          featured: row[:featured])
-
-        GC.start if total_running_count % 5000 == 0
-
-        if total_running_count % 1000 == 0
-          Rails.logger.debug "#{total_running_count} in #{distance_of_time_in_words_to_now(start_time, true)}"
-          @cap_import_logger.info total_running_count.to_s + ' in ' + distance_of_time_in_words_to_now(start_time, true)
-        end
-      rescue => e
-        @cap_import_logger.error "Problem with #{row.inspect}"
-        @cap_import_logger.error e.inspect << "\n" << e.backtrace.join("\n")
-      end
-    end
-    @cap_import_logger.info 'Finished import.' + Time.zone.now.to_s
-    @cap_import_logger.info lines.to_s + ' lines of file: ' + args.file_location.to_s + ' were processed.'
-    @cap_import_logger.info total_running_count.to_s + 'records were processed in ' + distance_of_time_in_words_to_now(start_time, true)
-  end
```

Comment:
OK, `create_pubs_from_delta` noted for deprecation in https://github.com/sul-dlss/sul_pub/issues/356#issuecomment-341251365